magnum_cluster_api/resources.py

```diff
@@ -296,6 +296,15 @@ def get_object(self) -> pykube.ConfigMap:
         if cinder.is_enabled(self.cluster):
             volume_types = osc.cinder().volume_types.list()
             default_volume_type = osc.cinder().volume_types.default()
+            # Default is set in accordance to the CSI:
+            # https://github.com/kubernetes/cloud-provider-openstack/blob/d228854cf58e7b4ed93d5e7ba68ab639450e3449/docs/cinder-csi-plugin/using-cinder-csi-plugin.md#supported-parameters
+            # If allow_availability_zone_fallback is set to False in Cinder
+            # and "nova" AZ is not present in Cinder, CSI request will fail.
+            volume_availability_zone = "nova"
+            if not CONF.cinder.cross_az_attach:
+                volume_availability_zone = self.cluster.labels.get(
+                    "availability_zone", volume_availability_zone
+                )
```

Comment:
I feel like this is making an assumption that the default AZ is `nova`.

I think what we should instead do is if `[cinder]/cross_az_attach` is set to `false`, we should set the availablity zone -- otherwise, we leave it empty and maintain the existing behaviour.

Reply:
Ok, sure will do. 

Just thought that matching the default value in csi should provide same behavior and simplify a resulting dict update which should be done conditionally then (as I don't think I can set the value to `None` here - it's gonna change the behavior indeed).